src/main/kotlin/lotto/model/LottoStore.kt

```diff
@@ -1,9 +1,19 @@
 package lotto.model
 
-class LottoStore(private val price: Money = Money.thousand) {
-    fun buy(money: Money): LottoTickets {
-        val ticketCount = (money / price).toInt()
+import lotto.model.number.LottoNumbers
 
-        return LottoTickets.issue(ticketCount)
+class LottoStore(private val generator: NumbersGenerator) {
+    fun buy(moneyAmount: Money, listOfCandidateNumbers: List<LottoNumbers> = listOf()): LottoTickets {
+        val manualTickets = LottoTickets(listOfCandidateNumbers.map { LottoTicket(it) })
+
+        return manualTickets + LottoTickets.issue(
+            calculateAutoTicketCount(moneyAmount, manualTickets.size),
+            generator
+        )
```

Comment:
아래의 코드로 개선해 보면 어떨까요?

```suggestion
        return manualTickets + LottoTickets(
            (1..calculateAutoTicketCount(moneyAmount, manualTickets.size)).map {
                LottoTicket(LottoNumbers(generator.getNumbers(LottoNumber.MAXIMUM, LottoNumbers.CANDIDATE_SIZE)))
            }
        )
```